tests/common.go

```diff
@@ -28,21 +28,25 @@ import (
 
 const (
 	// defaultSpecsRoot specifies the default location of the base specs directory in the Torpedo container
-	defaultSpecsRoot     = "/specs"
-	schedulerCliFlag     = "scheduler"
-	nodeDriverCliFlag    = "node-driver"
-	storageDriverCliFlag = "storage-driver"
-	specDirCliFlag       = "spec-dir"
-	logLocationCliFlag   = "log-location"
-	scaleFactorCliFlag   = "scale-factor"
+	defaultSpecsRoot                   = "/specs"
+	schedulerCliFlag                   = "scheduler"
+	nodeDriverCliFlag                  = "node-driver"
+	storageDriverCliFlag               = "storage-driver"
+	specDirCliFlag                     = "spec-dir"
+	logLocationCliFlag                 = "log-location"
+	scaleFactorCliFlag                 = "scale-factor"
+	storageDriverUpgradeVersionCliFlag = "storage-driver-upgrade-version"
+	storageDriverBaseVersionCliFlag    = "storage-driver-base-version"
 )
 
 const (
-	defaultScheduler      = "k8s"
-	defaultNodeDriver     = "ssh"
-	defaultStorageDriver  = "pxd"
-	defaultLogLocation    = "/mnt/torpedo_support_dir"
-	defaultAppScaleFactor = 10
+	defaultScheduler                   = "k8s"
+	defaultNodeDriver                  = "ssh"
+	defaultStorageDriver               = "pxd"
+	defaultLogLocation                 = "/mnt/torpedo_support_dir"
+	defaultAppScaleFactor              = 10
+	defaultStorageDriverUpgradeVersion = "1.2.11.6"
```

Comment:
The defaults should be defined in the storage driver. This version is Portworx specific. The tests can pass in an empty version string as the default one and the storage driver can then react on it and use the default ones.

Reply:
The storage driver implementation should not worry about the default upgrade version required for the test. 
Here we are using 'pxd' as the default storage driver, so shouldn't it be okay to specify default to be px specific version? 
It is similar to when the user gives 'k8s' as the scheduler, they have to give the proper spec dir for that scheduler. 
What do you think?

Reply:
Setting the default storage driver/scheduler/node-driver here makes sense because this piece of code instantiates them.
But setting some specific driver's constants here is a bit fuzzy. Let the storage driver decide what their default versions are. In this way even for a new storage driver the upgrade tests will work even if the input args are not specified.. I don't think we set the default k8s directory here?